src/SFML/Graphics/Sprite.cpp

```diff
@@ -61,22 +63,23 @@ Sprite::Sprite(const Texture& texture, const IntRect& rectangle) : m_texture(nul
 void Sprite::setTexture(const Texture& texture, bool resetRect)
 {
     // Recompute the texture area if requested, or if there was no valid texture & rect before
-    if (resetRect || (!m_texture && (m_textureRect == sf::IntRect())))
+    if (resetRect || (!m_texture && (m_textureRect == IntRect())))
     {
         setTextureRect(IntRect({0, 0}, Vector2i(texture.getSize())));
     }
 
-    // Assign the new texture
     m_texture = &texture;
 }
 
 
 ////////////////////////////////////////////////////////////
 void Sprite::setTextureRect(const IntRect& rectangle)
 {
-    if (rectangle != m_textureRect)
+    IntRect absoluteRect(rectangle.getAbsoluteRect());
```

Comment:
Note that a negative sized textureRect is a valid use case to flip vertically or horizontally the displayed texture area.

Reply:
Oh